src/models/inbound/dfspConsentRequests.model.ts

```diff
@@ -0,0 +1,281 @@
+  get pubSub (): PubSub {
+    return this.config.pubSub
+  }
+
+  get dfspBackendRequests (): DFSPBackendRequests {
+    return this.config.dfspBackendRequests
+  }
+
+  get thirdpartyRequests (): ThirdpartyRequests {
+    return this.config.thirdpartyRequests
+  }
+
+  async onValidateRequest (): Promise<void> {
+    const { request, toParticipantId } = this.data
+
+    try {
+      const response = await this.dfspBackendRequests.validateConsentRequests(request)
+
+      if (!response) {
+        throw new Errors.MojaloopFSPIOPError(
+          null as unknown as string,
+          null as unknown as string,
+          null as unknown as string,
+          Errors.MojaloopApiErrorCodes.TP_FSP_CONSENT_SCOPES_ERROR
+        )
+      }
+
+      if (!response.isValid) {
+        throw new Errors.MojaloopFSPIOPError(
+          null as unknown as string,
+          null as unknown as string,
+          null as unknown as string,
+          Errors.MojaloopApiErrorCodes.TP_NO_SUPPORTED_SCOPE_ACTIONS
+        )
+      }
+
+      this.data.response = response
+      if (response.authChannels.includes('WEB')) {
+        const webConsentRequestResponse: tpAPI.Schemas.ConsentRequestsIDPutResponseWeb = {
+          scopes: request.scopes,
+          callbackUri: request.callbackUri,
+          authChannels: ['WEB'],
+          authUri: response.authUri!,
+          initiatorId: toParticipantId
+        }
+        await this.thirdpartyRequests.putConsentRequests(request.id, webConsentRequestResponse, toParticipantId)
+      } else {
+        const otpConsentRequestResponse: tpAPI.Schemas.ConsentRequestsIDPutResponseOTP = {
+          scopes: request.scopes,
+          callbackUri: request.callbackUri,
+          authChannels: ['OTP'],
+          initiatorId: toParticipantId
+        }
+        await this.thirdpartyRequests.putConsentRequests(request.id, otpConsentRequestResponse, toParticipantId)
+      }
+
+    } catch (error) {
+      const mojaloopError = this.reformatError(error)
+
+      this.logger.push({ error }).error('start -> RequestIsValid')
+      this.logger.push({ mojaloopError }).info(`Sending error response to ${toParticipantId}`)
+
+      await this.thirdpartyRequests.putConsentRequestsError(
+        request.id,
+        mojaloopError as unknown as fspiopAPI.Schemas.ErrorInformationObject,
+        toParticipantId
+      )
+
+      // throw error to stop state machine
+      throw error
+    }
+  }
+
+  async onStoreReqAndSendOTP (): Promise<void> {
+    const { request, response } = this.data
+
+    try {
+      if (response?.authChannels.includes('WEB')) {
+        await this.dfspBackendRequests.storeConsentRequests(request)
+      } else {
+        await this.dfspBackendRequests.sendOTP(request)
+      }
+
+    } catch (error) {
+      this.logger.push({ error }).error('RequestIsValid -> success')
+      // throw error to stop state machine
+      throw error
+    }
+  }
+
+  reformatError (err: Error): Errors.MojaloopApiErrorObject {
```

Comment:
Does this belong elsewhere, like utils or something?

Reply:
Ok moved it to separate file shared/util